app/views/devise/two_factor_authentication/show.html.erb

```diff
@@ -7,5 +11,9 @@
   <%= submit_tag "Submit" %>
 <% end %>
 
-<%= link_to 'Resend Code', resend_code_user_two_factor_authentication_path, action: :get %>
-<%= link_to "Sign out", destroy_user_session_path, :method => :delete %>
\ No newline at end of file
+<% if current_user.direct_otp %>
+<%= link_to "Resend Code", resend_code_user_two_factor_authentication_path, action: :get %>
+<% else %>
+<%= link_to "Send me a code", resend_code_user_two_factor_authentication_path, action: :get %>
+<% end %>
```

Comment:
Why is the else statement necessary? Resending a code only applies to the direct OTP option, right? If people were sending an SMS to a user while using the TOTP option, they were not using the app properly, so I don't think we should maintain backwards compatibility with that. I'm fine with making this a breaking change and making this version 2.0.